test/unit/kubernetes-deploy/kubernetes_resource/daemon_set_test.rb

```diff
@@ -38,22 +38,99 @@ def test_deploy_passes_when_updated_available_does_match
       "observedGeneration": 2,
     }
 
-    ds_template = build_ds_template(status: status)
-    ds = build_synced_ds(template: ds_template)
+    ds_template = build_ds_template(filename: 'daemon_set.yml', status: status)
+    ds = build_synced_ds(ds_template: ds_template)
     assert_predicate(ds, :deploy_succeeded?)
   end
 
+  def test_deploy_passes_when_ready_pods_for_one_node
+    status = {
+      "desiredNumberScheduled": 1,
+      "updatedNumberScheduled": 1,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', status: status)
+    pod_templates = load_fixtures(filenames: ['daemon_set_pod.yml'])
+    node_templates = load_fixtures(filenames: ['node.yml'])
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: pod_templates, node_templates: node_templates)
+    assert_predicate(ds, :deploy_succeeded?)
+  end
+
+  def test_deploy_passes_when_ready_pods_but_node_added
+    status = {
+      "desiredNumberScheduled": 1,
+      "updatedNumberScheduled": 1,
+      "numberReady": 0,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', status: status)
+    pod_templates = load_fixtures(filenames: ['daemon_set_pod_not_ready.yml'])
+    node_templates = load_fixtures(filenames: ['node.yml'])
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: pod_templates, node_templates: node_templates)
+    refute_predicate(ds, :deploy_succeeded?)
+
+    node_added_status = {
+      "desiredNumberScheduled": 3,
+      "updatedNumberScheduled": 3,
+      "numberReady": 1,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', status: node_added_status)
+    pod_templates = load_fixtures(filenames: ['daemon_set_pod.yml'])
```

Comment:
For realism, shouldn't this return three pods, two of which are not ready? I think that right now, this test would still pass if the check were `@pods.all?(&:ready)`, unfiltered by node.

Reply:
I changed numberReady to 2, and made the fixture have 2 ready pods.